app/graphql/mutations/digital_object/update_projects.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module Mutations
+  module DigitalObject
+    class UpdateProjects < Mutations::BaseMutation
+      argument :id, ID, required: true
+      argument :primary_project, Inputs::StringKey, required: true
+      argument(
+        :other_projects,
+        [Inputs::StringKey],
+        "An array representing the new set of other projects for the given object, "\
+        "or an empty array to remove all projects but the primary.",
+        required: true
+      )
+
+      field :digital_object, Types::DigitalObjectInterface, null: true
+      field :user_errors, [Types::Errors::FieldedInput], null: false
+
+      def resolve(id:, primary_project:, other_projects:)
```

Comment:
If primary_project and other_projects are each optional for this mutation, the logic below would have to change to accommodate that.